### PR TITLE
W-15591992: Upgrade log4j version 2.17.1 to fix security vulnerability

### PR DESCRIPTION
**Pull request by dhairyashah-sf**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -68,6 +68,30 @@
...
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-slf4j-impl</artifactId>
+                <version>2.17.1</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-1.2-api</artifactId>
+                <version>2.17.1</version>
+            </dependency>
         </dependencies>
```
Comment (aby-philip-sf) [inline pom.xml:95]: Please ensure all these were dependencies before too. If it is not, we don't have to bring it in.

File: pom.xml [hunk 2]
```diff
@@ -68,6 +68,30 @@
...
                 <version>2.10.1</version>
             </dependency>
+            <!--
+                Upgrade log4j version to fix security vulnerability
+                ref - https://logging.apache.org/log4j/2.x/security.html
+            -->
+            <dependency>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-api</artifactId>
+                <version>2.17.1</version>
```
Comment (aby-philip-sf) [inline pom.xml:78]: Can define.a <log4j.version> under <properties> and use it.

**Review by aby-philip-sf — COMMENTED** [review]
comments added

File: pom.xml (hunk 1 above)
Comment (dhairyashah-sf) [reply pom.xml:95]: As discussed offline, these are declared in dependencyManagement and we just specify the version to be used for dependencies as well as transitive dependencies.

File: pom.xml (hunk 2 above)
Comment (dhairyashah-sf) [reply pom.xml:78]: noted

**Review by aby-philip-sf — APPROVED** [review]

**Pull request merged**